services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/OpenHouseInternalRepositoryImpl.java

```diff
@@ -163,16 +163,27 @@ public TableDto save(TableDto tableDto) {
   }
 
   private boolean skipEligibilityCheck(
-      Map<String, String> existingTableProps, Map<String, String> newTableprops) {
+      Map<String, String> existingTableProps, Map<String, String> newTableProps) {
+    // If on the same cluster, table update is primary -> primary and must check all keys
+    if (existingTableProps
+        .get(getCanonicalFieldName(CLUSTER_ID))
+        .equals(newTableProps.get(getCanonicalFieldName(CLUSTER_ID)))) {
+      return false;
+    }
+
+    // If the table is explicitly marked as replicated, skip eligibility check
+    if (Boolean.parseBoolean(
```

Comment:
are we planning to use this property to skip SE and retention jobs as well? What is the plan for that?

Reply:
The presumption is that the destination table does not exist when the active replica is created, so this is treated as a new table with backfilled data. So the original table is the SoT for all the policies